ScriptureRenderingPipeline/Rendering.cs

```diff
@@ -26,68 +28,202 @@ public static class Rendering
+                    };
                 }
-            }
 
-            var output = renderer.Render(document);
-            string outputFilePath = Path.Join(repoDir, "output.docx");
-            using (var stream = new FileStream(outputFilePath, FileMode.Create))
-            {
-                output.Write(stream);
+                if (fileType == "pdf")
+                {
+                    var latexConverterUrl = Environment.GetEnvironmentVariable("PDFConversionEndpoint");
+                    var fontMappingUrl = Environment.GetEnvironmentVariable("FontLookupLocation");
+                    if (string.IsNullOrEmpty(latexConverterUrl))
+                    {
+                        return GenerateErrorAndLog($"PDF conversion was requested but converter url was not specified", log, 400);
+                    }
+
+                    if (string.IsNullOrEmpty(fontMappingUrl))
+                    {
+                        return GenerateErrorAndLog($"PDF conversion was requested but font lookup file url was not specified", log, 400);
+                    }
+
+                    // Find out which font we should use in the pdf
+                    var fonts = await GetFonts(fontMappingUrl);
+                    var config = CreateLatexConfig(req.Query);
+                    config.Font = SelectFontForDocument(document, fonts);
+                    log.LogInformation($"Selected {config.Font} for rendering");
+
+                    LatexRenderer renderer = new LatexRenderer(config);
+                    var result = renderer.Render(document);
+                    HttpClient client = new HttpClient();
+                    var pdfResult = await client.PostAsync(latexConverterUrl, new StringContent(result));
+                    if (!pdfResult.IsSuccessStatusCode)
+                    {
+                        return GenerateErrorAndLog("Error rendering pdf", log);
+                    }
+                    var stream = await pdfResult.Content.ReadAsStreamAsync();
+                    return new FileStreamResult(stream, "application/octet-stream")
+                    {
+                        FileDownloadName = fileName ?? "output.pdf"
+                    };
+                }
+
+                if (fileType == "latex")
+                {
+                    var fontMappingUrl = Environment.GetEnvironmentVariable("FontLookupLocation");
+                    if (string.IsNullOrEmpty(fontMappingUrl))
+                    {
+                        return GenerateErrorAndLog($"Latex conversion was requested but font lookup file url was not specified", log, 400);
+                    }
+
+                    // Find out which font we should use in the latex file
+                    var fonts = await GetFonts(fontMappingUrl);
+                    var config = CreateLatexConfig(req.Query);
+                    config.Font = SelectFontForDocument(document, fonts);
+                    log.LogInformation($"Selected {config.Font} for rendering");

```

Comment:
This font selection code looks identical between pdf and latex.  Consider pulling it out into a function.